employee-rostering-frontend/src/ui/components/EditableComponent.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+
+
+export interface EditableComponentProps {
+  viewer: JSX.Element;
+  editor: JSX.Element;
+}
+
+export interface EditableComponentState {
+  isEditing: boolean;
+}
+
+export class EditableComponent extends React.Component<EditableComponentProps, EditableComponentState> {
+  constructor(props: EditableComponentProps) {
+    super(props)
+    this.state = {isEditing: false};
+  }
+
+  startEditing() {
+    this.setState({...this.state, isEditing: true});
```

Comment:
Component shouldn't allow to change its state from outside. If you change `isEditing` to a prop, you'll no longer need to keep the EditableComponent's `ref` in order to invoke this method. You can simple pass the `isEditing` flag to it as a prop derived from the state of it's parent component, the DataTable.